server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -141,6 +143,17 @@ public final class IndexModule {
         Property.NodeScope
     );
 
+    /**
+     * Index setting which used to determine how the data is cached locally fully or partially
+     */
+    public static final Setting<DataLocalityType> INDEX_STORE_LOCALITY_SETTING = new Setting<>(
+        "index.store.data_locality",
+        DataLocalityType.FULL.name(),
+        DataLocalityType::getValueOf,
+        Property.IndexScope,
```

Comment:
this needs to be a dynamic setting for tiering from hot to warm, can we update it to dynamic and a Private setting?

Reply:
We can take this up when we have migration support ready from hot to warm and vice versa